velox/expression/CastExpr.cpp

```diff
@@ -145,10 +145,21 @@ VectorPtr castToDate(
       return castResult;
     }
     case TypeKind::TIMESTAMP: {
-      auto* inputVector = input.as<SimpleVector<Timestamp>>();
       static const int32_t kSecsPerDay{86'400};
+      auto inputVector = input.as<SimpleVector<Timestamp>>();
+      const auto& queryConfig = context.execCtx()->queryCtx()->queryConfig();
+      auto sessionTzName = queryConfig.sessionTimezone();
+      using AdjustForTZFunc = std::function<void(Timestamp&)>;
+      AdjustForTZFunc adjustTimestampForTimezone = [&](Timestamp& input) {};
+      if (queryConfig.adjustTimestampToTimezone() && !sessionTzName.empty()) {
+        auto* timeZone = date::locate_zone(sessionTzName);
+        adjustTimestampForTimezone = [&](Timestamp& input) {
+          input.toTimezone(*timeZone);
+        };
+      }
       context.applyToSelectedNoThrow(rows, [&](int row) {
         auto input = inputVector->valueAt(row);
+        adjustTimestampForTimezone(input);
```

Comment:
Callback is usually worse than a check against a const boolean.

Reply:
Agree. I will revert the change.